### PR TITLE
[FEATURE] Add try_increase_bin_number_to

### PR DESCRIPTION
**Pull request by eseiler**
It's handy for the dynamic IBF because I need to differentiate whether increasing the number of bins is costly (resizing).

It also seems like a useful API to have; manually determining whether a resize happens is not trivial and requires some knowledge about the IBF.

### CODE REVIEW

**Comment by seqan-actions** [comment]
Documentation preview available at https://docs.seqan.de/preview/seqan/hibf/253

File: include/hibf/interleaved_bloom_filter.hpp [hunk 1]
```diff
@@ -300,8 +300,20 @@ class interleaved_bloom_filter : private seqan::hibf::bit_vector
                 (*this)[bin.value + offset] = 0;
     }
 
+    /*!\brief Tries increasing the number of bins stored in the Interleaved Bloom Filter.
+     * \param[in] new_bin_count The new number of bins.
+     * \returns `true` if the number of bins was set, `false` if the number of bins was not set.
+     * \sa seqan::hibf::interleaved_bloom_filter::increase_bin_number_to
+     *
+     * If increasing the number of bins would require a resize, the function returns `false` and
```
Comment (eseiler) [inline include/hibf/interleaved_bloom_filter.hpp:308]: Also add warning about agents being invalidated.

File: include/hibf/interleaved_bloom_filter.hpp [hunk 2]
```diff
@@ -300,8 +300,20 @@ class interleaved_bloom_filter : private seqan::hibf::bit_vector
                 (*this)[bin.value + offset] = 0;
     }
 
+    /*!\brief Tries increasing the number of bins stored in the Interleaved Bloom Filter.
```
Comment (smehringer) [inline include/hibf/interleaved_bloom_filter.hpp:303]:
```suggestion
    /*!\brief Tries increasing the number of bins stored in the Interleaved Bloom Filter without reallocating memory.
```

File: include/hibf/interleaved_bloom_filter.hpp [hunk 3]
```diff
@@ -300,8 +300,20 @@ class interleaved_bloom_filter : private seqan::hibf::bit_vector
...
+     * \param[in] new_bin_count The new number of bins.
+     * \returns `true` if the number of bins was set, `false` if the number of bins was not set.
+     * \sa seqan::hibf::interleaved_bloom_filter::increase_bin_number_to
+     *
+     * If increasing the number of bins would require a resize, the function returns `false` and
+     * does not increase the number of bins.
+     * If old and new bin count are the same, `true` is returned.
+     * If the new bin count is smaller than the old bin count, `false` is returned.
```
Comment (smehringer) [inline include/hibf/interleaved_bloom_filter.hpp:311]:
```suggestion
     * If the new bin is bigger than the current bin count:
     * - and if a resize is required, returns `false` and nothing happens
     * - and if no resize is necessary, sets the old bin count to the new and return `true`
     * If old and new bin count are the same, `true` is returned.
     * If the new bin count is smaller than the old bin count, `false` is returned.
```

**Review by smehringer — APPROVED** [review]

**Pull request merged**